### PR TITLE
Time zone menu fix

### PR DESCRIPTION
**Pull request by radekkaluzik**
### Description
`<Dropdown>` replaced by `<Menu>` - fixed menu functionality (problems described in bug ticket).
Fixed Alert message - it only got called once at render when the value was undefined - function refactored to useMemo hook.

[RHCLOUD-32296](https://issues.redhat.com/browse/RHCLOUD-32296)

---

#### Before:
image
image
image

#### After:
image
image
image

---

### Checklist ☑️
- [ x ] PR only fixes one issue or story 
- [ x ] Change reviewed for extraneous code 
- [ x ] UI best practices adhered to 
- [ x ] Commits squashed and meaningfully named 
- [ x ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

**Comment by radekkaluzik** [comment]
Functionality verified with UTC Time Zone Converter
https://dateful.com/convert/utc

File: src/components/Notifications/TimeConfig.tsx [hunk 1]
```diff
@@ -51,17 +54,26 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
-    if (getTimePreference.status === 200) {
-      return getTimePreference.payload?.value as string;
+    if (getTimePreference.error) {
+      return undefined;
     }
 
-    return undefined;
-  }, [getTimePreference.payload?.value, getTimePreference.status]);
+    return getTimePreference.payload?.value as string;
+  }, [getTimePreference.payload?.value, getTimePreference.error]);
```
Comment (fhlavac) [inline src/components/Notifications/TimeConfig.tsx:70]:
```suggestion
const timePref = useMemo(
    () =>
      getTimePreference.error
        ? undefined
        : (getTimePreference.payload?.value as string),
    [getTimePreference.payload?.value, getTimePreference.error]
  );
```

File: src/components/Notifications/TimeConfig.tsx [hunk 2]
```diff
@@ -51,17 +54,26 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
+
+  const timeconfigTitle = useMemo(() => {
+    return `Any daily digest emails you've opted into will be sent at ${
+      timeSelect?.utcTime ? timeSelect?.utcTime : '00:00'
+    } UTC`;
+  }, [timeSelect?.utcTime]);
```
Comment (fhlavac) [inline src/components/Notifications/TimeConfig.tsx:76]:
```suggestion
  const timeconfigTitle = useMemo(() => `Any daily digest emails you've opted into will be sent at ${
      timeSelect?.utcTime ? timeSelect?.utcTime : '00:00'
    } UTC`, [timeSelect?.utcTime]);
```

**Comment by fhlavac** [comment]
Tested locally and it looks good!

**Review by karelhala — APPROVED** [review]

**Pull request merged**